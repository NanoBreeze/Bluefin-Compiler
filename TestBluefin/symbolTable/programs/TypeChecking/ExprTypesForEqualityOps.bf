int x;
int main() {
	float  y;
	if (x==y) {	}
	else { x != y; }
	bool z;
	if (x == y == z) {}
}