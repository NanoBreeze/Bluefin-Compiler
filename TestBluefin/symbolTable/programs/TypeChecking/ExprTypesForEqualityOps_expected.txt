x - t_int - t_float
y - t_float - t_float
x==y - t_bool - t_bool
x - t_int - t_float
y - t_float - t_float
x!=y - t_bool - t_bool
x - t_int - t_float
y - t_float - t_float
x==y - t_bool - t_bool
z - t_bool - t_bool
x==y==z - t_bool - t_bool
